docs/reference-implementation/Troubleshooting.md

```diff
@@ -81,6 +81,35 @@ Deployment of service [HealthService | BackgroundProcessor | CatalogService] fai
 
 **Solution:** Re-run the failing step - there's a high probability that it will work. If not on second run, investigate pod health (look at AKS logs, potential error messages from deployment etc.).
 
+---
+
+**Error:** Deploy CatalogService workload / Install workload CatalogService on AKS clusters failed
+
+```console
+certificate for catalogservice-ingress-secret pending.. 
+Deployment not ready. Retrying... 1/5
+..
+Deployment not ready. Retrying... 5/5
+##[error]Script failed with exit code: 1
+```
+
+**Description:** The "Install workload CatalogService on AKS clusters" step actively monitors the provisioning of a LetsEncrypt certificate for the ingress. If the certificate is not ready, or cannot successfully provisioned this task will fail.
+
+**Solution:** This is most of the time caused by hitting LetsEncrypt thresholds (see [Rate Limits](https://letsencrypt.org/docs/rate-limits/) for more details) for certificate provisioning. You can try to re-run the step, or wait for the certificate to be ready.
+
+```console
+# Check certificate status
+kubectl get certificates -n workload
+
+# Check certificate request status
+kubectl get certificaterequests -n workload
```

Comment:
how about adding a "k delete certificaterequest" as a possible solution? this should immediately trigger reissue

Reply:
I was thinking about that as well - what's important though is to make sure that the reason is clear - in case you're hitting a rate limit on the Let's Encrypt side you can delete it as often as you want and nothing will change.

And I'd suggest to delete the certificate not the certificaterequest. Let me rephrase it a bit.